Zend/tests/final_properties/access/property_assignment_combined1.phpt

```diff
@@ -0,0 +1,106 @@
+--TEST--
+Test that final properties can't be mutated by combined assignment operators
+--XFAIL--
+??= doesn't work
+--FILE--
+<?php
+
+class Foo
+{
+    final public int $property1 = 1;
+    final public string $property2 = "";
+}
+
+$foo = new Foo();
+
+try {
+    $foo->property1 += 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 -= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 *= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 /= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 **= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 &= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 |= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 ^= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 <<= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property1 >>= 1;
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property2 .= "foo";
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+try {
+    $foo->property2 ??= "foo";
```

Comment:
I haven't managed to make this case work.

Reply:
This only does an assignment if the property is null. So it's correct that this doesn't error as-is, you'd have to use a different default value.

Reply:
What a little bastard :'D But thanks for the info, it was too difficult for me to discover how it really works.